lib/apipie/validator.rb

```diff
@@ -177,23 +177,23 @@ def description
     class EnumValidator < BaseValidator
       def initialize(param_description, argument)
         super(param_description)
-        @array = argument
+        @argument = argument
       end
 
       def validate(value)
-        @array.include?(value)
+        values.include?(value)
       end
 
       def self.build(param_description, argument, options, proc)
-        self.new(param_description, argument) if argument.is_a?(Array)
+        self.new(param_description, argument) if argument.is_a?(Array) || argument.respond_to?(:call)
```

Comment:
@mathieujobin I'm not sure how to disambiguate this from the `ProcValidator` here:
https://github.com/Apipie/apipie-rails/blob/eeef4de93ac7fe68942cb0e878a9e8353eceeb17/lib/apipie/validator.rb#L313

If you have any thoughts, I'd love to hear them.

I thought about introducing a class that somehow wraps the callable, but it feels rather heavy.

Reply:
Because EnumValidator will receive a Proc as well ? 
Can you check that is it an Enum before? 

`.respond_to?(:call)` can be a lot of things... so, you accept to build a Enum validator with either an Array or a Proc? not an Enum ? is a Enum a Proc underneath?

it would be good to add a new test where the confusion between the two validator can be demonstrated and the right one is being selected. this will help finding a working implementation.